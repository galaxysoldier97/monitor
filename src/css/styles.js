import { typography } from '@material-ui/system';

const styles = {
  navigation: {
    paddingBottom: 15,
  },
  title: {
    fontSize: 24,
    fontWeight: typography.fontWeightLight,
    marginBottom: 20,
  },
  paper: {
    padding: 30,
  },
  clear: {
    clear: 'both',
  },
};

export default styles;
