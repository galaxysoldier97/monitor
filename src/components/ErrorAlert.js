import React from 'react';
import { Button, Grid, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';
import RedColor from '@material-ui/core/colors/red';
import PropTypes from 'prop-types';
import { t } from 'mt-react-library/functions';
import Box from '@material-ui/core/Box';

const useStyles = makeStyles(theme => ({
  root: {
    margin: 'auto',
    width: '100%',
    backgroundColor: RedColor[50],
    padding: theme.spacing(2, 3),
    borderRadius: 5,
  },
  title: {
    color: RedColor[600],
    fontWeight: 'bold',
    textShadow: '0 0 5px white',
  },
}));

export const ErrorAlert = ({message, marginY = 5, fullWidth = false, retry}) => {
  const classes = useStyles();

  return (
    <Box marginY={marginY} margin="auto" style={fullWidth ? {width: '100%'} : {}}>
      <div className={classes.root} style={fullWidth ? {width: '100%'} : {maxWidth: 500}}>
        <Grid container justify="space-between" alignItems="center">
          <Grid item>
            <Typography className={classes.title}>{t('error.title')}</Typography>
            <Typography>{message}</Typography>
          </Grid>
          {retry && (
            <Grid item>
              <Button onClick={retry} variant="outlined">{t('error.retry')}</Button>
            </Grid>
          )}
        </Grid>
      </div>
    </Box>
  );
};

ErrorAlert.propTypes = {
  message: PropTypes.string.isRequired,
  marginY: PropTypes.number,
  fullWidth: PropTypes.bool,
  retry: PropTypes.func,
};
