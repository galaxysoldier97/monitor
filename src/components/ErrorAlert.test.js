import React from 'react';
import renderer from 'react-test-renderer';
import { ErrorAlert } from './ErrorAlert';
import { createMuiTheme } from '@material-ui/core';
import { ThemeProvider } from '@material-ui/core/styles';
import { IntlProvider } from 'react-intl';
import EN from '../i18n/en.json';

test('ErrorAlert snapshot', () => {
  const theme = createMuiTheme({});
  const component = renderer.create(
    <IntlProvider locale="en" messages={EN}>
      <ThemeProvider theme={theme}>
        <ErrorAlert message="test" />
      </ThemeProvider>,
    </IntlProvider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
