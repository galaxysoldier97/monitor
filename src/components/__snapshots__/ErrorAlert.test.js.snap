// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert snapshot 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-4"
    style={Object {}}
  >
    <div
      className="makeStyles-root-1"
      style={
        Object {
          "maxWidth": 500,
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <p
            className="MuiTypography-root makeStyles-title-2 MuiTypography-body1"
          >
            An error occurred
          </p>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            test
          </p>
        </div>
      </div>
    </div>
  </div>,
  ",",
]
`;
